### PR TITLE
Add unit 11: Kerberos ticket policy

### PR DESCRIPTION
**Pull request by abbra**
Kerberos ticket policy unit describes ways of control of Kerberos
tickets in FreeIPA with the help of ticket policies.

### CODE REVIEW

File: 11-kerberos-ticket-policy.rst [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+- `Unit 3: User management and Kerberos authentication <3-user-management.rst>`_
+
+In this module you will explore how to manage properties of Kerberos
+authentication and authorization in FreeIPA.
+
+Kerberos ticket policies
+------------------------
+
+FreeIPA's primary authentication mechanism is based on Kerberos infrastructure.
+Each user has associated Kerberos principal and potential aliases. Each FreeIPA
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: has an associated

File: 11-kerberos-ticket-policy.rst [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+service has own Kerberos service and, optionally, alias names as well. Depending
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: has its own

File: 11-kerberos-ticket-policy.rst [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+on who initiates a communication flow, in Kerberos terminology there are client
+and a target (server) principals. Any Kerberos principal can be used in both
+client and server roles, though typically FreeIPA services are used as targets
+and users are used as clients.
+
+A client first authenticates to a Key Distribution Center and obtains a ticket
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: add (KDC) after Key Distribution Center

File: 11-kerberos-ticket-policy.rst [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+granting ticket (TGT). With the ticket granting ticket in possession, the client
+then asks a KDC for a service ticket to the target Kerberos principal. After the
+service ticket was issued by the KDC, the client can initiate its
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: was -> is

File: 11-kerberos-ticket-policy.rst [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+place because individual parts of it are applied to different objects and at
+different stages of authentication and authorization processes. Instead, it is
+split into several parts which are associated with the corresponding Kerberos
+principals, whether they are used as clients or as targets.
+
+For users, Kerberos ticket life time and renewal age can be managed with
+Kerberos ticket policy commands described in ``ipa help krbtpolicy`` manual.
+
+If no specific policy is associated with a user, a default one is applied. To
+manage default policy the same ``ipa krbtpolicy-*`` commands are used, without
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: manage the default

File: 11-kerberos-ticket-policy.rst [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+explicit user name.
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: an explicit user name.

File: 11-kerberos-ticket-policy.rst [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+
+Display the current Kerberos ticket policy::
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: Do you want to be clear that this is the default policy?

File: 11-kerberos-ticket-policy.rst [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+
+  [admin@client]$ kinit admin
+  Password for admin@IPADEMO.LOCAL:
+  [admin@client]$ ipa krbtpolicy-show
+    Max life: 86400
+    Max renew: 604800
+
+Reset the policy to the default::
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: This resets the default policy to the default :-)

Maybe "installation default"?

File: 11-kerberos-ticket-policy.rst [hunk 9]
```diff
@@ -0,0 +1,387 @@
...
+
+Reset per-user policy for user ``admin``::
+
+  [admin@client]$ ipa krbtpolicy-reset admin
+    Max life: 28800
+    Max renew: 86640
+
+Currently (FreeIPA 4.8.4), FreeIPA does not allow to associate a custom Kerberos
+ticket policy with a Kerberos service principal. As result, only default
+Kerberos ticket policy is applicable to service principals.
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: to associating -> associating

or "FreeIPA does not allow a Kerberos service principal to have a custom Kerberos ticket policy."

File: 11-kerberos-ticket-policy.rst [hunk 10]
```diff
@@ -0,0 +1,387 @@
...
+   on FAST. SPAKE implementation in FreeIPA currently only supports a single
+   factor authentication.
+
+In the context of authentication indicators, FAST and SPAKE pre-authentication
+methods give higher level of protection than an exchange using encrypted
+timestamp method, traditional for Kerberos 5.
+
+Each authentication indicator conveys the fact that KDC was able to
+pre-authenticate the initial ticket granting ticket exchange using chosen
+mechanism. This fact can further be used to differentiate issued ticket life
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: differentiate the issued life...

File: 11-kerberos-ticket-policy.rst [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+time and renewal age.
+
+With FreeIPA 4.8.3 or later, Kerberos ticket policy allows to set different life
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: allows one or allows an administrator or allows a user

File: 11-kerberos-ticket-policy.rst [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+As result, ``admin`` user can now use its ticket up to 8 hours and renew it
+until 48 hours since the original ticket was obtained.
+
+Enforcing authentication indicators
+-----------------------------------
+
+Authentication indicators from the ticket granting ticket are copied by the KDC
+into service tickets issued with the help of the TGT presented by a Kerberos
+client. The indicators can be seen by the applications receiving a communication
+encrypted with the service ticket. This allows to permit restricted access to
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: allows who? An administrator?

File: 11-kerberos-ticket-policy.rst [hunk 13]
```diff
@@ -0,0 +1,387 @@
...
+
+At the moment, there are no known applications that implement authentication
+indicator-based authorization. Instead, FreeIPA provides a check for an
+authentication indicator at KDC side. This means that a lack of a specific
+authentication indicator in TGT can result in denying an issuance of a requested
+service ticket. A consequence is that an application will never see any user
+with a ticket that does not contain a specified authentication indicator.
+
+In order to enable the check, add authentication indicator to a service using
+``ipa service-mod`` command. We can create a new service and associate
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst:264]: the pkinit

File: 11-kerberos-ticket-policy.rst [hunk 14]
```diff
@@ -0,0 +1,387 @@
...
+   Modified service "my-service/client.ipademo.local@IPADEMO.LOCAL"
+   ------------------------------------------------------
+     Principal name: my-service/client.ipademo.local@IPADEMO.LOCAL
+     Principal alias: my-service/client.ipademo.local@IPADEMO.LOCAL
+     Managed by: client.ipademo.local
+
+Authentication indicators and FreeIPA services
+----------------------------------------------
+
+Authentication indicators can become an effective way to enforce use of a
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: enforce the use of a...

File: 11-kerberos-ticket-policy.rst [hunk 15]
```diff
@@ -0,0 +1,387 @@
...
+particular pre-authentication method. However, there are caveats. Since
+KDC-based enforcement does not allow anyone to obtain a service ticket to a
+Kerberos service if they do not posses an authentication indicator in question,
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: posses -> possess

File: 11-kerberos-ticket-policy.rst [hunk 16]
```diff
@@ -0,0 +1,387 @@
...
+a great care has to be used when assigning authentication indicators to internal
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: a great care -> great care

File: 11-kerberos-ticket-policy.rst [hunk 17]
```diff
@@ -0,0 +1,390 @@
...
+client. The indicators can be seen by the applications receiving a communication
+encrypted with the service ticket. This allows an application administrator to
+permit restricted access to only those clients who used specific
+pre-authentication mechanisms to obtain their initial ticket granting ticket.
+For example, an application might decide to only allow access to a specialized
+resource to people who used smart-card authentication initially, even if the
+application itself only supports Kerberos authentication.
+
+At the moment, there are no known applications that implement authentication
+indicator-based authorization. Instead, FreeIPA provides a check for an
```
Comment (frozencemetery) [inline 11-kerberos-ticket-policy.rst:257]: Should we link to how to do that?  I'm thinking of http://web.mit.edu/kerberos/krb5-latest/doc/appdev/gssapi.html#gssapi-authind-attr

**Review by frozencemetery — COMMENTED** [review]
In general looks good; one suggestion inline.

File: 11-kerberos-ticket-policy.rst (hunk 17 above)
Comment (abbra) [reply 11-kerberos-ticket-policy.rst:257]: The target audience of the workshop are system administrators, I don't think it makes sense to add it here.

File: 11-kerberos-ticket-policy.rst [hunk 18]
```diff
@@ -0,0 +1,390 @@
...
+and users are used as clients.
+
+A client first authenticates to a Key Distribution Center (KDC) and obtains a
+ticket granting ticket (TGT). With the ticket granting ticket in possession, the
+client then asks a KDC for a service ticket to the target Kerberos principal.
+After the service ticket is issued by the KDC, the client can initiate its
+application-specific communication to the target application server.
+
+When a KDC issues a ticket, there are few properties that can be controlled with
+the help of a Kerberos policy in FreeIPA. Each ticket has own life time and a
```
Comment (rcritten) [inline 11-kerberos-ticket-policy.rst]: has its own

**Review by rcritten — COMMENTED** [review]
Looks good, just two more suggestions.

**Comment by rcritten** [comment]
LGTM

**Pull request merged**